tests/unit/test_credentials.py

```diff
@@ -3128,6 +3134,61 @@ def test_can_pass_basic_auth_token(self):
         self.assertEqual(creds.token, 'token')
         self.assertEqual(creds.method, 'container-role')
 
+    def test_can_pass_auth_token_from_file(self):
+        token_file_path = self.tempdir + '/token.jwt'
+        token_file = open(token_file_path, "x")
+        token_file.write("Basic auth-token")
+        token_file.close()
+        environ = {
+            'AWS_CONTAINER_CREDENTIALS_FULL_URI': 'http://localhost/foo',
+            'AWS_CONTAINER_AUTHORIZATION_TOKEN_FILE': token_file_path,
+        }
+        fetcher = self.create_fetcher()
+        timeobj = datetime.now(tzlocal())
+        timestamp = (timeobj + timedelta(hours=24)).isoformat()
+        fetcher.retrieve_full_uri.return_value = {
+            "AccessKeyId": "access_key",
+            "SecretAccessKey": "secret_key",
+            "Token": "token",
+            "Expiration": timestamp,
+        }
+        provider = credentials.ContainerProvider(environ, fetcher)
+        creds = provider.load()
+
+        fetcher.retrieve_full_uri.assert_called_with(
+            'http://localhost/foo',
+            headers={'Authorization': 'Basic auth-token'},
+        )
+        self.assertEqual(creds.access_key, 'access_key')
+        self.assertEqual(creds.secret_key, 'secret_key')
+        self.assertEqual(creds.token, 'token')
+        self.assertEqual(creds.method, 'container-role')
+
+    def test_throws_error_on_invalid_token_file(self):
+        token_file_path = '/some/path/token.jwt'
+        environ = {
+            'AWS_CONTAINER_CREDENTIALS_FULL_URI': 'http://localhost/foo',
+            'AWS_CONTAINER_AUTHORIZATION_TOKEN_FILE': token_file_path,
+        }
+        fetcher = self.create_fetcher()
+        provider = credentials.ContainerProvider(environ, fetcher)
+
+        with self.assertRaises(FileNotFoundError):
+            provider.load()
+
+    def test_throws_error_on_illegal_header(self):
+        environ = {
+            'AWS_CONTAINER_CREDENTIALS_FULL_URI': 'http://localhost/foo',
+            'AWS_CONTAINER_AUTHORIZATION_TOKEN': "invalid\r\ntoken",
+        }
+        fetcher = self.create_fetcher()
+        provider = credentials.ContainerProvider(environ, fetcher)
+
+        with self.assertRaisesRegex(
+            ValueError, "Auth token value is not a legal header value"
+        ):
```

Comment:
I'm not sure we need to test the exact error message here. Just asserting the right error gets raised should be sufficient.